doc/development.md

```diff
@@ -35,24 +35,46 @@ Create virtual environment for vdsm:
 
 ## Building Vdsm
 
-To configure sources (run `./configure --help` to see configuration options):
+To configure sources (create Makefiles and configure script): 
 
     git clean -xfd
     ./autogen.sh --system --enable-timestamp
-    make
 
-To test Vdsm (refer to tests/README for further tests information):
+(It is recommended to start fresh after an update, from the previous step, before
+building RPMs, because the generated Makefile also contains the release number.)
 
-    make check
+Now you can create the virtual environment (https://docs.python.org/3/library/venv.html),
+which is necessary to run the tests later. This needs to be done only once:
+
+    make venv
+
+To see configuration options run `./configure -h`. Configure things, if necessary,
+then start make to build Vdsm:
+
+    make
 
-To create an RPM:
+To create the RPMs:
 
     make rpm
 
 To upgrade your system with local build's RPM:
 
     make upgrade
 
+To test Vdsm you need to activate the virtual environment first.
+(refer to [tests/README.md](tests/README.md) for further testing information):
+
+    . ~/.venv/vdsm/bin/activate
+
+Note how the prompt gets a `(vdsm)` prefix.
+After that you can:
+
+    make check
+
+To leave the virtual environment:
+
+    deactivate
+
```

Comment:
Testing vdsm needs a new section. This section is about building vdsm.

Reply:
So we should not mention the venv here, but only in the tests/README.md?

Reply:
We need to add a new section in this document about running the tests. Entering the virtual env is not related to building vdsm.

Reply:
We need all basic information for developers in one place. tests/README can handle stuff which is rarely needed, or converted to other documents in doc/.

Reply:
I added a basic test example in a seperate section, including the link to the tests/README